extensions.go

```diff
@@ -233,6 +235,69 @@ func (c *Client) GetCurrentNet() (wire.BitcoinNet, error) {
 	return c.GetCurrentNetAsync().Receive()
 }
 
+// FutureGetHeadersResult is a future promise to delivere the result of a
+// getheaders RPC invocation (or an applicable error).
+//
+// NOTE: This is a btcsuite extension ported from
+// github.com/decred/dcrrpcclient.
+type FutureGetHeadersResult chan *response
+
+// Receive waits for the response promised by the future and returns the
+// getheaders result.
+//
+// NOTE: This is a btcsuite extension ported from
+// github.com/decred/dcrrpcclient.
+func (r FutureGetHeadersResult) Receive() ([]wire.BlockHeader, error) {
+	res, err := receiveFuture(r)
+	if err != nil {
+		return nil, err
+	}
+
+	// Unmarshal result as a slice of strings.
+	var result []string
+	err = json.Unmarshal(res, &result)
+	if err != nil {
+		return nil, err
+	}
+
+	// Deserialize the []string into []wire.BlockHeader.
+	headers := make([]wire.BlockHeader, len(result))
+	for i, headerHex := range result {
+		serialized, err := hex.DecodeString(headerHex)
+		err = headers[i].Deserialize(bytes.NewReader(serialized))
+		if err != nil {
+			return nil, err
+		}
+	}
+	return headers, nil
+}
+
+// GetHeadersAsync returns an instance of a type that can be used to get the result
+// of the RPC at some future time by invoking the Receive function on the returned instance.
+//
+// See GetHeaders for the blocking version and more details.
+//
+// NOTE: This is a btcsuite extension ported from
+// github.com/decred/dcrrpcclient.
+func (c *Client) GetHeadersAsync(blockLocators []chainhash.Hash, hashStop *chainhash.Hash) FutureGetHeadersResult {
+	locators := make([]string, len(blockLocators))
+	for i := range blockLocators {
+		locators[i] = blockLocators[i].String()
+	}
+	cmd := btcjson.NewGetHeadersCmd(locators, hashStop.String())
```

Comment:
This will crash if a `nil` is passed for the stop hash.  I think this should use the same pattern as `GetBlockHeader`.

```Go
	hash := ""
	if hashStop != nil {
		hash = hashStop.String()
	}
	cmd := btcjson.NewGetHeadersCmd(locators, hash)
```